app/black_spots/views.py

```diff
@@ -74,3 +76,136 @@ def get_queryset(self):
+        param_val = self.get_required_param(key, request)
+        try:
+            param_val = int(param_val)
+        except ValueError:
+            raise ParseError(detail="The '{}' parameter must be an integer".format(key))
+        return param_val
+
+    def get_required_int(self, key, request):
+        """
+        Helper for getting a required int URL parameter value or raising a ParseError
+        :param key:  URL parameter key
+        :param request:  The request object
+        """
+        param_val = self.get_required_param(key, request)
+        try:
+            param_val = int(param_val)
+        except ValueError:
+            raise ParseError(detail="The '{}' parameter must be an integer".format(key))
+        return param_val
+
+    def get_required_dt_param(self, key, request):
+        """
+        Helper for getting a required datetime URL parameter value or raising a ParseError
+        :param key:  URL parameter key
+        :param request:  The request object
+        """
+        param_val = self.get_required_param(key, request)
+        return parse_and_validate_dt(param_val, key)
+
+    def choose_assignments(self, assignments, num_personnel, shift_start, shift_end):
+        """
+        Select the assignments according to the supplied parameters
+        :param assignments: filtered queryset of assignments (black spots)
+        :param num_personnel: number of assignments to choose
+        :param shift_start: start dt of the shift
+        :param shift_end: end dt of the shift
+        """
+
+        # The multiplier used for determining the set of assignments to choose from. The size of
+        # the list of possible assignments is determined by multiplying this number by the number of
+        # personnel. A higher number will result in a greater variance of assignments among shifts.
+        fuzz_factor = 4
+
+        # Create a set of assignments with the highest forecasted severity score to sample from.
+        assignments = assignments.order_by('-severity_score')[:num_personnel * fuzz_factor]
+
+        # Specify a random seed based on the shift, so assignments are deterministic during the same
+        # shift, yet vary from shift to shift.
+        random.seed(hash('{}_{}'.format(shift_start, shift_end)))
+
+        # Return the sampled list of assignments
+        return random.sample(assignments, num_personnel)
+
+    def scale_by_toddow(self, assignments, shift_start, shift_end):
+        """
+        Scale the expected load forecast (severity score) by the time of day and day of week
+        :param assignments: filtered queryset of assignments (black spots)
+        :param shift_start: start dt of the shift
+        :param shift_end: end dt of the shift
+        """
+
+        # TODO: use time of day/day of week to generate the appropriate scaling factor
+        scaling_factor = 1 / (7 * 24.0)
+
+        # Scale the severity score by the scaling factor
+        for assignment in assignments:
```

Comment:
This could perhaps be shortened slightly by replacing it with a call to `map` but it's not critical.